compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/transform/StatefulDoFnTransform.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (C) 2018 Seoul National University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nemo.compiler.frontend.beam.transform;
+
+import org.apache.beam.runners.core.*;
+import org.apache.beam.runners.core.construction.SerializablePipelineOptions;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.reflect.DoFnInvoker;
+import org.apache.beam.sdk.transforms.reflect.DoFnInvokers;
+import org.apache.beam.sdk.util.WindowedValue;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.sdk.values.WindowingStrategy;
+import org.apache.nemo.common.ir.OutputCollector;
+import org.apache.nemo.common.ir.vertex.transform.Transform;
+import org.apache.nemo.compiler.frontend.beam.NemoPipelineOptions;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Stateful DoFn transform implementation.
+ * This implementation supposes that the inputs are KV.
+ * We then process the inputs that have the same key in the same doFnRunner for stateful processing.
+ *
+ * @param <K> key type.
+ * @param <InputT> input type.
+ * @param <OutputT> output type.
+ */
+public final class StatefulDoFnTransform<K, InputT, OutputT> implements
```

Comment:
This class is being used only in DoTransformTest#testStatefulFn, and much of its code is same as `SimpleDoFnTransform`.

Would it be possible to merge the two classes into one, or introduce an abstract parent class containing the common code?

Reply:
This class is for `StatefulDoFn` but current PR does not fully support the `StatefulDoFn`. I will remove this class in this PR.

Reply:
Created an issue for `StatefulDoFn`: https://issues.apache.org/jira/projects/NEMO/issues/NEMO-219